### PR TITLE
7013 add blinded blocks block contents type

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Add blinded block and blinded block contents deseriliasation types

[Spec reference here](https://github.com/ethereum/beacon-APIs/pull/302)

## Fixed Issue(s)

#7013 

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by mehdi-aouadi** [comment]
This PR is rebased on https://github.com/ConsenSys/teku/pull/7146

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java [hunk 1]
```diff
@@ -16,70 +16,159 @@
...
-                  SignedBlockContents.isInstance,
+                  SignedBlockContents.isSignedBlockContentsInstance,
                   s -> s.contains("blob_sidecars"),
-                  SIGNED_BLOCK_CONTENTS_TYPE)
+                  SignedBlockContentsSchema.create(
+                          spec.getGenesisSpecConfig().toVersionDeneb().orElseThrow(),
+                          SignedBlobSidecarSchema.create(
+                              BlobSidecarSchema.create(
+                                  new BlobSchema(
+                                      spec.getGenesisSpecConfig().toVersionDeneb().orElseThrow()))),
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java]: Basically this, it's just generally not fully exposed here...
if you look at `GetLightClientBootstrap` as an example, but basically you'd be doing the .create elsewhere, and here it'd be something like
```
SchemaDefinitionsDeneb.required(
                schemaDefinitionCache.getSchemaDefinition(SpecMilestone.DENEB))
            .getSignedBlockContentsSchema()
            .getJsonTypeDefinition()
```

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java [hunk 2]
```diff
@@ -16,70 +16,159 @@
...
+                      .getSignedBeaconBlockSchema()
+                      .getJsonTypeDefinition())
+              .withType(
+                  SignedBlindedBlockContents.isSignedBlindedBlockContentsInstance,
+                  s -> s.contains("blob_sidecars"),
+                  SignedBlindedBlockContentsSchema.create(
+                          spec.getGenesisSpecConfig().toVersionDeneb().orElseThrow(),
+                          SignedBlindedBlobSidecarSchema.create(BlindedBlobSidecarSchema.create()),
+                          spec.getGenesisSchemaDefinitions().getSignedBeaconBlockSchema())
+                      .getJsonTypeDefinition())
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java]:
```
SchemaDefinitionsDeneb.required(
                schemaDefinitionCache.getSchemaDefinition(SpecMilestone.DENEB))
            .getSignedBlindedBlockContentsSchema()
            .getJsonTypeDefinition()
```
ish..

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java]: admittedly arugable either way in an actual test, but in the endpoints it'd look like this...

**Review by rolfyone — APPROVED** [review]
Lgtm

**Pull request merged**